### PR TITLE
Decorate_a_Class

### PR DESCRIPTION
**Pull request by AJ193**
### Project requirements

- Created a class Nameable
1. Implemented a method called correct_name that will raise a NotImplementedError.
2. Person class inherits from Nameable
3. This class has a method correct_name implemented. It should simply return the name attribute.

- Prepare base Decorator
1. It inherits from Nameable.
2. In the constructor assign a nameable object from params to an instance variable.
3. Implemented the correct_name method that returns the result of the correct_name method of the @nameable.

### Prepare CapitalizeDecorator and TrimmerDecorator
- For the CapitalizeDecorator:
1. Create a class that inherits from the base Decorator class.
2. Implement a method correct_name that capitalizes the output of @nameable.correct_name.

- For the TrimmerDecorator:
1. Create a class that inherits from the base Decorator class.
2. Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum 
    of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: nameable.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Nameable
```
Comment (gemmen29) [inline nameable.rb:1]: - [x] Kindly make sure that the `Person` class inherits from `Nameable` class, then make sure that this class has a method `correct_name` implemented as per those [requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_decorator_v2.md#:~:text=Turn%20your%20Person,the%20name%20attribute.).

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @AJ193, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@ class Person < Nameable
   attr_reader :id
 
   def initialize(age, name = 'Unknown', parent_permission: true)
-    super(name)
+    super
```
Comment (julie-ify) [inline person.rb]: - [x] Kindly change this to `super()`. I could not run your app due to this. That should fix all your issues.

  Note that `super` and `super()` have very different applications. You can see [this StackOverflow comment](https://stackoverflow.com/a/31816281) for more description.

**Review by julie-ify — CHANGES_REQUESTED** [review]
Hi @AJ193,

Good job so far!

## Highlights!
- Good code organization and decorator ✔️
- Descriptive README ✔️ 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by muneebulrehman — APPROVED** [review]
Hi @AJ193 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🥳
Congratulations! 🎉 

# Status: Approved✔️

## Highlights
- No linter errors
- Descriptive PR
- Good follow-up with the previous reviewers.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**